src/content/guides/tree-shaking.md

```diff
@@ -7,19 +7,28 @@ contributors:
   - alexjoverm
   - avant1
   - MijaelWatts
+  - dmitriid
 related:
   - title: Tree shaking with webpack 2, TypeScript and Babel
     url: https://alexjoverm.github.io/2017/03/06/Tree-shaking-with-Webpack-2-TypeScript-and-Babel/
   - title: Tree-shaking with webpack 2 and Babel 6
     url: http://www.2ality.com/2015/12/webpack-tree-shaking.html
   - title: webpack 2 Tree Shaking Configuration
     url: https://medium.com/modus-create-front-end-development/webpack-2-tree-shaking-configuration-9f1de90f3233#.15tuaw71x
+  - title: Issue 2867
+    url: https://github.com/webpack/webpack/issues/2867
+  - title: Issue 4784
+    url: https://github.com/webpack/webpack/issues/4784
 ---
 
 _Tree shaking_ is a term commonly used in the JavaScript context for dead-code elimination. It relies on the [static structure](http://exploringjs.com/es6/ch_modules.html#static-module-structure) of ES2015 module syntax, i.e. [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) and [`export`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export). The name and concept have been popularized by the ES2015 module bundler [rollup](https://github.com/rollup/rollup).
 
 The webpack 2 release came with built-in support for ES2015 modules (alias _harmony modules_) as well as unused module export detection.
 
+**Note**: Webpack doesn't perform tree-shaking by itself. It relies on third party tools like [UglifyJS](/plugins/uglifyjs-webpack-plugin/) to perform actual dead code elimination. There are situations where tree-shaking may not effective. For example, when the tool cannot guarantee that a particular code path doesn't lead to side-effects, this code may remain in the generated bundle even if you are sure it shouldn't.
```

Comment:
@dmitriid It's a very fussy one, but as far as I can tell, the correct form of 'webpack' is always with a lowercase 'w', even when it feels wrong :)

@skipjack Can you confirm / deny?

Reply:
Yeah it as per the [branding guidelines](https://github.com/webpack/media), it should always be lowercase. It is tough though in certain scenarios, honestly I just try to avoid using it at the beginning of sentences. Usually there's a way to rephrase without reducing the quality of the sentence but I'm not sure in this case.